core/server/api/schedules.js

```diff
@@ -35,21 +35,34 @@ exports.publishPost = function publishPost(object, options) {
         function (cleanOptions) {
             cleanOptions.status = 'scheduled';
 
-            return apiPosts.read(cleanOptions)
-                .then(function (result) {
-                    post = result.posts[0];
-                    publishedAtMoment = moment(post.published_at);
+            return dataProvider.Base.transaction(function (transacting) {
+                cleanOptions.transacting = transacting;
 
-                    if (publishedAtMoment.diff(moment(), 'minutes') > publishAPostBySchedulerToleranceInMinutes) {
-                        return Promise.reject(new errors.NotFoundError(i18n.t('errors.api.job.notFound')));
-                    }
+                // CASE: apiPosts.read and apiPosts.edit happen in a transaction, signalise `forUpdate` to knex
```

Comment:
In my opinion all operations in the endpoint have to run in the same single transaction. This single transaction is created in the scheduler api (^ line 38) and forwarded for any case. 

As soon as we signalise `forUpdate` (happens in `apiPosts.read`), the row is locked and e.g. a client request to change the same post, would have to wait until the scheduler finished to publish the post. This avoids:

1. collision errors for internal operations
2. lost updates